tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -689,9 +689,17 @@ private static Trace getOrCreateCurrentTrace() {
 
     @VisibleForTesting
     static void clearCurrentTrace() {
+        logClearingTrace();
         currentTrace.remove();
         MDC.remove(Tracers.TRACE_ID_KEY);
         MDC.remove(Tracers.TRACE_SAMPLED_KEY);
         MDC.remove(Tracers.REQUEST_ID_KEY);
     }
+
+    private static void logClearingTrace() {
+        log.debug("Clearing current trace", SafeArg.of("maybeTrace", Optional.ofNullable(currentTrace.get())));
```

Comment:
nit: Let's wrap this with log.isDebugEnabled to prevent optional and safeArg allocations when debug logging is disabled.

note: no reason to use an Optional for the safe-arg, as far as the logging system is concerned empty optional and null are indistinguishable.

Reply:
While we're here, thoughts on mirroring the debug line after a traceId is set? That way we get a full sense of the scope transitions.